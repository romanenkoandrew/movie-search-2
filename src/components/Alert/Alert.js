import React from 'react';
import PropTypes from 'prop-types';
import { Alert } from 'antd';
import styles from './Alert.styles';

const AlertMessage = ({ description, type, closeAlert }) => {
  React.useEffect(() => {
    setTimeout(closeAlert, 3000);
  }, []);
  return (
    <Alert
      css={styles.alert}
      message={type ? 'Error' : 'Success'}
      description={description}
      type={type ? 'error' : 'success'}
      onClose={closeAlert}
      closable
      showIcon
    />
  );
};

AlertMessage.propTypes = {
  description: PropTypes.string.isRequired,
  type: PropTypes.bool.isRequired,
  closeAlert: PropTypes.func.isRequired,
};
export default AlertMessage;
