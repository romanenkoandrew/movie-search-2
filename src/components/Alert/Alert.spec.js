import React from 'react';
import { shallow } from 'enzyme';
import Alert from './Alert';

describe('Alert component', () => {
  it('should render Alert component with type=true', () => {
    const component = shallow(
      <Alert description='Success' type={true} closeAlert={() => {}} />
    );
    expect(component).toMatchSnapshot();
  });
  it('should render Alert component with type=false', () => {
    const component = shallow(
      <Alert description='Success' type={false} closeAlert={() => {}} />
    );
    expect(component).toMatchSnapshot();
  });
});
