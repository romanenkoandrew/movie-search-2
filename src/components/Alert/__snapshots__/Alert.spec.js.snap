// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component should render Alert component with type=false 1`] = `
<EmotionCssPropInternal
  __EMOTION_LABEL_PLEASE_DO_NOT_USE__="AlertMessage"
  __EMOTION_TYPE_PLEASE_DO_NOT_USE__={[Function]}
  closable={true}
  css={[Function]}
  description="Success"
  message="Success"
  onClose={[Function]}
  showIcon={true}
  type="success"
/>
`;

exports[`Alert component should render Alert component with type=true 1`] = `
<EmotionCssPropInternal
  __EMOTION_LABEL_PLEASE_DO_NOT_USE__="AlertMessage"
  __EMOTION_TYPE_PLEASE_DO_NOT_USE__={[Function]}
  closable={true}
  css={[Function]}
  description="Success"
  message="Error"
  onClose={[Function]}
  showIcon={true}
  type="error"
/>
`;
